File: backend/src/main/kotlin/org/gymapp/backend/mapper/GymMapper.java
package org.gymapp.backend.mapper;

import org.gymapp.backend.model.Gym;
import org.gymapp.library.response.GymDto;
import org.mapstruct.Mapper;

import java.util.List;

@Mapper(componentModel = "spring")
public interface GymMapper {

    GymDto modelToDto(Gym gym);

    List<GymDto> modelsToDtos(List<Gym> gyms);

}

File: backend/src/main/kotlin/org/gymapp/backend/mapper/GymMapper.kt
package org.gymapp.backend.mapper

import org.gymapp.backend.model.Gym
import org.gymapp.library.response.GymDto
import org.mapstruct.Mapper

@Mapper(componentModel = "spring")
interface GymMapper {

    fun modelToDto(gym: Gym?): GymDto

    fun modelsToDtos(gyms: List<Gym?>): List<GymDto>
}